### PR TITLE
Build Docker images

### LINKED ISSUE #260: Automatic regression test via Docker Hub
A few times we have experienced that Labgrid got broken for our use. Thus we came up with this idea.

Setup a docker hub project like this one https://hub.docker.com/r/deifwpt/labgrid-client/~/settings/automated-builds/ which is linked to a repository.

This is our own, but it is only linked to our own repository where we put our Dockerfile. If the Docker file is moved to labgrid, then any changes in labgrid/master will cause a new docker build.

This docker build, can then be used to do nightly regression tests in our CI system.

We would not use the image for our "production" tests. We would setup a separate docker image and manually rebuild that when we need the latest labgrid.

This of course means that the dockerfiles made by https://github.com/krevsbech  would be moved to labgrid

### PR DESCRIPTION
**Pull request by esben**
This add Dockerfile for building Docker images including the Labgrid distributed infrastructure components, and integrates build testing of them in Travis CI.

Replaces #284
Closes #260

### CODE REVIEW

**Comment by esben** [comment]
@krevsbech will follow up with a PR with Docker compose file example when this PR is merged.

File: docker/README.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+All 3 images are to be considered base images
+with the required software installed.
+No policy or configuration is done.
+
+
+labgrid-coordinator usage
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The labgrid-coordinator comes a preconfigured Crossbar.io server.
```
[the remark this answers is not in the document]
Comment (esben) [reply docker/README.rst]: Fixing

File: docker/README.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  .. code-block:: bash
+
+     $ docker run -t -p 20408:20408 -v $HOME/crossbar:/opt/crossbar
+	 labgrid-coordinator
+
+
+labgrid-client usage
+~~~~~~~~~~~~~~~~~~~~
+
+The labgrid-client image can be to used to
```
[the remark this answers is not in the document]
Comment (esben) [reply docker/README.rst]: Fixing

File: docker/client/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+RUN mkdir -p /opt/labgrid
+COPY ./ /opt/labgrid/
+RUN cd /opt/labgrid \
+ && pip install -r requirements.txt \
+ && python setup.py install
+
+RUN apt-get update -q \
+ && apt-get install -q -y microcom \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
```
Comment (henriknj) [inline docker/client/Dockerfile]: Have you considered using mutli-stage builds?

**Pull request merged**